packages/cinterop/src/commonMain/kotlin/io/realm/interop/errors/RealmCoreErrors.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.interop.errors
+
+/**
+ * Generic representation of a Realm-Core exception.
+ */
+open class RealmCoreException(message: String?) : Exception(message)
```

Comment:
Here everything is turned into Exceptions. In Realm Java we signal some things as Errors. I don't have strong preferences, either way, and maybe throwing Errors is a bit too strong. Not sure if you thought about this?